vision/detect_image_properties.rb

```diff
@@ -12,55 +12,67 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def detect_image_properties project_id:, image_path:
+require "uri"
+
+def detect_image_properties image_path:
   # [START vision_image_property_detection]
-  # project_id = "Your Google Cloud project ID"
   # image_path = "Path to local image file, eg. './image.png'"
-  
+
   require "google/cloud/vision"
 
-  vision = Google::Cloud::Vision.new project: project_id
-  image  = vision.image image_path
+  image_annotator = Google::Cloud::Vision::ImageAnnotator.new
 
-  image.properties.colors.each do |color|
-    puts "Color #{color.red}, #{color.green}, #{color.blue}"
+  response = image_annotator.image_properties_detection(
+    image: image_path
+  )
+
+  response.responses.each do |res|
+    res.image_properties_annotation.dominant_colors.colors.each do |color|
+      puts "Color #{color.color.red}, #{color.color.green}, #{color.color.blue}"
```

Comment:
The stuttering looks kind of bad. I think it would help if you called the variable `color_info` instead of `color` (to accurately reflect its type). That will make the code make a bit more sense. For example:

```ruby
res.image_properties_annotation.dominant_colors.colors.each do |color_info|
  color = color_info.color
  puts "Color #{color.red}, #{color.green}, #{color.blue}"
end
```

Reply:
```ruby
res.image_properties_annotation.dominant_colors.colors.map(&:color).each do |color|
  puts "Color #{color.red}, #{color.green}, #{color.blue}"
end
```